### PR TITLE
partially reverts reaction chamber

### PR DESCRIPTION
**Pull request by Time-Green**
Partially reverts #57071 

## Changelog
:cl:
balance: Temporarily reverts the changes made to the plumbing reaction chamber untill minichem gets added (never)
/:cl:

Why:
I simply underestimated how big this change would be. You could do everything you normally could, but the space limitations were just absolutely massive. At least until minichem gets added, it's better to leave it a little broken

Yes, I could make some other way to controle reactions that would be less broken and cost less space, but I'm just so fucking tired of dealing with the balance implications of plumbing. I just wanna add stupid shit and pat myself on the back

### CODE REVIEW

**Comment by Time-Green** [comment]
If it knows so fucking well how the indentation is supposed to work then it can fucking indent the tgui itself

**Comment by stylemistake** [comment]
How about you enable eslint in vscode or use tgui.bat to fix these things.

**Comment by Time-Green** [comment]
> How about you enable eslint in vscode or use tgui.bat to fix these things.

I figured out —fix is a thing, which fixed it

File: code/datums/components/plumbing/reaction_chamber.dm [hunk 1]
```diff
@@ -18,10 +18,18 @@
 	if(RC.emptying)
 		return
 
-	process_request(amount = min(MACHINE_REAGENT_TRANSFER, RC.target_volume - reagents.total_volume), dir = dir)
-
-	if(RC.target_volume > round(reagents.total_volume, CHEMICAL_VOLUME_ROUNDING)) //not enough yet
-		return
+	for(var/RT in RC.required_reagents)
```
Comment (Mothblocks) [inline code/datums/components/plumbing/reaction_chamber.dm:21]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/datums/components/plumbing/reaction_chamber.dm [hunk 2]
```diff
@@ -18,10 +18,18 @@
...
+		var/has_reagent = FALSE
+		for(var/A in reagents.reagent_list)
+			var/datum/reagent/RD = A
```
Comment (Mothblocks) [inline code/datums/components/plumbing/reaction_chamber.dm:24]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Also, use `as anything`.

File: code/modules/plumbing/plumbers/reaction_chamber.dm [hunk 3]
```diff
@@ -80,12 +82,17 @@
 /obj/machinery/plumbing/reaction_chamber/ui_data(mob/user)
 	var/list/data = list()
 
+	var/list/text_reagents = list()
+	for(var/A in required_reagents) //make a list where the key is text, because that looks alot better in the ui than a typepath
+		var/datum/reagent/R = A
```
Comment (Mothblocks) [inline code/modules/plumbing/plumbers/reaction_chamber.dm:87]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Also, use `as anything`.

File: code/modules/plumbing/plumbers/reaction_chamber.dm [hunk 4]
```diff
@@ -6,8 +6,10 @@
 	buffer = 200
 	reagent_flags = TRANSPARENT | NO_REACT
 
-	///At what volume do we start reacting?
-	var/target_volume = 200
+	/**list of set reagents that the reaction_chamber allows in, and must all be present before mixing is enabled.
+	* example: list(/datum/reagent/water = 20, /datum/reagent/fuel/oil = 50)
```
Comment (Mothblocks) [inline code/modules/plumbing/plumbers/reaction_chamber.dm:10]:
```suggestion
	/**
	* list of set reagents that the reaction_chamber allows in, and must all be present before mixing is enabled.
	* example: list(/datum/reagent/water = 20, /datum/reagent/fuel/oil = 50)
```

I believe this is required for SpacemanDMM to check.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
Thanks for the review! I still need to get used to the new soulles naming conventions

File: code/datums/components/plumbing/reaction_chamber.dm [hunk 5]
```diff
@@ -9,26 +9,34 @@
...
 		return
 
-	process_request(amount = min(MACHINE_REAGENT_TRANSFER, RC.target_volume - reagents.total_volume), dir = dir)
-
-	if(RC.target_volume > round(reagents.total_volume, CHEMICAL_VOLUME_ROUNDING)) //not enough yet
-		return
+	for(var/required_reagent in chamber.required_reagents)
+		var/has_reagent = FALSE
+		for(var/containing_reagent_but_not_typecasted_yet in reagents.reagent_list)
+			var/datum/reagent/containg_reagent = containing_reagent_but_not_typecasted_yet as anything
```
Comment (Mothblocks) [inline code/datums/components/plumbing/reaction_chamber.dm:24]: Not how it works--here's the page on it: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists

**Comment by Time-Green** [comment]
I honestly wouldn't have blamed you if you repobanned me for that one
It's fixed now

**Pull request merged**